src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -80,28 +102,21 @@ const CharacterForm = ({ onClose }) => {
 
   useEffect(() => {
     if (!character) return
-    reset(prepareCharacterForFormReset(character))
+    reset(prepareCharacterForFormReset(character.person))
   }, [character])
 
   useEffect(() => {
-    if (isEdit && !character) {
-      router.push({
-        pathname: "/characters/[[...params]]",
-        query: undefined,
-      })
+    if (!getAllSpeciesLoading && !getAllHomeworldLoading) {
+      setIsLoadingFetch(false)
     }
-  }, [isEdit, character])
-
-  const dispatch = useDispatch()
+  }, [getAllSpeciesLoading, getAllHomeworldLoading])
```

Comment:
What's going on here?

Reply:
I removed the previous codes since now we have error messages to be displayed in cases where the character does not exist.

On another note, the new code has also been removed since I noticed there's no use in combining two loading states for species and homeworld options